src/components/Request/Pay/Pay.tsx

```diff
@@ -75,56 +87,59 @@ export const PayRequestLink = () => {
                 setLinkState('ALREADY_PAID')
                 return
             }
+            setLinkState('PAY')
+        } catch (error) {
+            console.error('Failed to fetch request link details:', error)
+        }
+    }
 
-            // Fetch token price
-            const tokenPrice = await utils.fetchTokenPrice(
-                requestLinkDetails.tokenAddress.toLowerCase(),
-                requestLinkDetails.chainId
-            )
-            tokenPrice && setTokenPrice(tokenPrice?.price)
+    useEffect(() => {
+        const pageUrl = typeof window !== 'undefined' ? window.location.href : ''
+        if (pageUrl) {
+            checkRequestLink(pageUrl)
+        }
+    }, [])
 
-            await fetchPointsEstimation(requestLinkDetails, tokenPrice)
+    useEffect(() => {
+        if (!requestLinkData) return
 
-            let recipientAddress = requestLinkDetails.recipientAddress
-            if (requestLinkDetails.recipientAddress.endsWith('eth')) {
-                recipientAddress = await utils.resolveFromEnsName(requestLinkDetails.recipientAddress.toLowerCase())
-            }
+        // Fetch token price
+        utils
+            .fetchTokenPrice(requestLinkData.tokenAddress.toLowerCase(), requestLinkData.chainId)
+            .then((tokenPriceData) => {
+                setTokenPriceData(tokenPriceData)
+            })
```

Comment:
Suggestion: if we can't fetch price, shall we set it to 0 in the UI? This might lead to an error when computing the x-chain fulfillment transaction, but then again squid has their own price engine so it shouldn't affect it

Reply:
the tokenPriceData will be undefined on failure, we can show an error